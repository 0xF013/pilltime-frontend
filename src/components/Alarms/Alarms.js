import React, { Component } from 'react';
import { connect } from 'react-redux';
import Alarm from '../Alarm';
import AlarmCreateForm from '../AlarmCreateForm';
import { loadAll } from '../../redux/alarms';

export class Alarms extends Component {
  componentDidMount() {
    this.props.dispatch(loadAll());
  }

  render() {
    const { items } = this.props;
    return (
      <div>
        <h2>Create alarm</h2>
        <AlarmCreateForm />
        <h2>Alarms</h2>
        { items.map(alarm => <Alarm key={alarm.id} alarm={alarm} />)}
      </div>
    );
  }
}

const mapStateToProps = state => state.alarms;

export default connect(mapStateToProps)(Alarms);