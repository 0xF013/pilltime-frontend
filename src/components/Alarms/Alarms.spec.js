import React from 'react';
import { Alarms } from './Alarms';
import { shallow } from 'enzyme';

describe('<Alarms />', () => {
  it('matches the snapshot', () => {
    const items = [
      { some: 'item1' },
      { some: 'item2' },
      { some: 'item3' },
    ];
    const wrapper = shallow(<Alarms items={ items } />);
    expect(wrapper).toMatchSnapshot();
  });
});