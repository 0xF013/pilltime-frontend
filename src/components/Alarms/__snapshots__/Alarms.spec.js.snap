exports[`<Alarms /> matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2>
        Create alarm
    </h2>
    <Connect(ReduxForm) />
    <h2>
        Alarms
    </h2>
    <Alarm
        alarm={
            Object {
                "some": "item1",
              }
        } />
    <Alarm
        alarm={
            Object {
                "some": "item2",
              }
        } />
    <Alarm
        alarm={
            Object {
                "some": "item3",
              }
        } />
</div>,
  "nodes": Array [
    <div>
      <h2>
            Create alarm
      </h2>
      <Connect(ReduxForm) />
      <h2>
            Alarms
      </h2>
      <Alarm
            alarm={
                  Object {
                        "some": "item1",
                      }
            } />
      <Alarm
            alarm={
                  Object {
                        "some": "item2",
                      }
            } />
      <Alarm
            alarm={
                  Object {
                        "some": "item3",
                      }
            } />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Alarms
        items={
                Array [
                        Object {
                          "some": "item1",
                        },
                        Object {
                          "some": "item2",
                        },
                        Object {
                          "some": "item3",
                        },
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Alarms {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "items": Array [
            Object {
              "some": "item1",
            },
            Object {
              "some": "item2",
            },
            Object {
              "some": "item3",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2>
                    Create alarm
          </h2>
          <Connect(ReduxForm) />
          <h2>
                    Alarms
          </h2>
          <Alarm
                    alarm={
                              Object {
                                        "some": "item1",
                                      }
                    } />
          <Alarm
                    alarm={
                              Object {
                                        "some": "item2",
                                      }
                    } />
          <Alarm
                    alarm={
                              Object {
                                        "some": "item3",
                                      }
                    } />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2>
                    Create alarm
          </h2>
          <Connect(ReduxForm) />
          <h2>
                    Alarms
          </h2>
          <Alarm
                    alarm={
                              Object {
                                        "some": "item1",
                                      }
                    } />
          <Alarm
                    alarm={
                              Object {
                                        "some": "item2",
                                      }
                    } />
          <Alarm
                    alarm={
                              Object {
                                        "some": "item3",
                                      }
                    } />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Alarms
    items={
        Array [
            Object {
              "some": "item1",
            },
            Object {
              "some": "item2",
            },
            Object {
              "some": "item3",
            },
          ]
    } />,
}
`;
